napari/layers/base/_base_key_bindings.py

```diff
@@ -3,7 +3,7 @@
 from napari.layers.base.base import Layer
 
 
-@Layer.bind_key(KeyCode.Space)
+@Layer.bind_key(KeyCode.Space, overwrite=True)
```

Comment:
In the OP
> Note: i set overwrite=True on the spacebar keybinding because it's such a core keybinding.

It's a pretty core keybind, common to every image/graphics program on my computer. We use overwrite for other such like control-z, control-c, etc.
So it seemed like a good idea to ensure it works. 🤷‍♂️

Reply:
Overwiritte is to remove the existing one. As far as I remember, it not prevent overwriting in different imports.

But I could be wrong.

Reply:
So maybe whole action should be migrated to standard shortcut manager? Manager itself prevent from overwritte.

Reply:
So the thing with that is it will get confusing. Each layer has a registered e.g. `register_points_mode_action` pan/zoom action and the keybind can be set in the shortcuts preferences (there is no `base` in the shortcuts dropdown.) But that's to ACTIVATE pan/zoom, like the button, while this spacebar action is the press-to-activate-release-to-restore.
But could register that and think of some naming for it...

Reply:
Thinking more, I really don't think this should be editable. 
It's just a feature: space activates pan/zoom when held down. 
Not every aspect needs to be customized I don't think.

Reply:
I don't think we should give up on customizability just because some technical issue is making it hard :P I would still prefer to leave this editable as everything else.

but I'm not really following the discussion about *why* this `overwrite` is necessary and/or doesn't work. Can't this be just a normal key binding like the rest?

Reply:
Here's what I have from the above discussion and my reading of the code:
- space for pan/zoom *while held* was never settable in the keybindings (and isn't in other apps I use), it's just AThingThatWorksTM
- I set overwrite here to try to ensure that behavior, but it doesn't do anything because the action manager hard codes overwrite on the keybinds from settings. Can drop it, whatever.
- Setting and unsetting Pan/Zoom is already a keybinding on every layer--not always the same one though!! (there's a PR to fix that but it's abandoned (https://github.com/napari/napari/pull/5021)
- Making this a `normal` keybinding in settings probably means trying to make it a viewer keybinding? Does that make sense? Adding it for every layer seems like asking for a lot of user confusion. Which is why again, I would advocate for the old behavior (which is what I tried to implement here, pure bugfix).

That's the executive summary I think?

Reply:
I would prefer it to be settable, but I'm definitely ok with it being a "viewer keybinding" rather than layer specific. In fact, I'm not super keen on layers having the same modes (panzoom, transform) but different shortcuts, but that's a discussion for another day :P

Either way, I dont't feel strongly about this, so if you think the "AThingThatWorksTM" approach, that's also fine :)